src/runtime/contrib/tensorrt/tensorrt_builder.cc

```diff
@@ -217,6 +231,20 @@ void TensorRTBuilder::CleanUp() {
   }
 }
 
+void TensorRTBuilder::AllocateDeviceBufferIfNeeded(nvinfer1::ICudaEngine* engine,
```

Comment:
We can just name it `AllocateDeviceBuffer` and add comments to mention we will bypass if the data entry is already on the GPU.